waddrmgr/manager.go

```diff
@@ -1820,3 +1875,103 @@ func Create(ns walletdb.ReadWriteBucket, seed, pubPassphrase, privPassphrase []b
 	// Use 48 hours as margin of safety for wallet birthday.
 	return putBirthday(ns, birthday.Add(-48*time.Hour))
 }
+
+// CreateWatchOnly creates a new watching-only address manager in the given namespace.
+//
+// No default account or scoped manager are created - it is up to the caller to create a new one
+// with NewAccountWatchingOnly and NewWatchingOnlyScopedKeyManager
+//
+// All public keys and information are protected by secret keys
+// derived from the provided public passphrases.  The public
+// passphrase is required on subsequent opens of the address manager.
+//
+// If a config structure is passed to the function, that configuration will
+// override the defaults.
+//
+// A ManagerError with an error code of ErrAlreadyExists will be returned the
+// address manager already exists in the specified namespace.
+func CreateWatchOnly(ns walletdb.ReadWriteBucket,
```

Comment:
Similar to the new scope key manager method, there's a lot of duplication here with the `Create` method, as a result I think the two of them can be heavily consolidated.

Reply:
Agreed. This will also make it possible to use table driven tests and reduce code duplication there. See comment further below.

Reply:
Fixed.